### PR TITLE
Kernel/syscalls: Require `inet` promise for AF_INET6 domain

### PR DESCRIPTION
**Pull request by logkos**
This goes in accordance with OpenBSD specification
https://man.openbsd.org/pledge.2#inet

### CODE REVIEW

**Review by nico — APPROVED** [review]
Thanks!

File: Kernel/Syscalls/socket.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@ namespace Kernel {
     do {                                          \
         if (domain == AF_INET)                    \
             TRY(require_promise(Pledge::inet));   \
+        else if (domain == AF_INET6)              \
```
[the remark this answers is not in the document]
Comment (logkos) [reply Kernel/Syscalls/socket.cpp:19]: I did not want to put all these slashes further away as the lint fails when they are misaligned and that would make It weird to look at

[the remark this answers is not in the document]
Comment (nico) [reply Kernel/Syscalls/socket.cpp:19]: 🤷

Not the best reason, but also doesn't matter all that much I suppose.

**Comment by ADKaster** [comment]
> This goes in accordance with OpenBSD specification

fwiw, our pledge() already doesn't follow the OpenBSD specification. We have some of our own pledges, and some of our pledges behave differently.  Nothing wrong with this change though, it maintains consistency with the surrounding code.

**Pull request merged**